api/apps/api/src/modules/projects/dto/update-project-tag.dto.ts

```diff
@@ -0,0 +1,25 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsNotEmpty, MaxLength, Validate } from 'class-validator';
+import { IsValidTagNameValidator } from '@marxan-api/modules/geo-feature-tags/validators/is-valid-tag-name.custom.validator';
+import {
+  tagMaxlength,
+  tagMaxLengthErrorMessage,
+} from '@marxan-api/modules/geo-feature-tags/dto/update-geo-feature-tag.dto';
+
+export class UpdateProjectTagDTO {
+  @ApiProperty()
+  @IsNotEmpty({ message: 'The Tag cannot not be empty' })
+  @Validate(IsValidTagNameValidator)
+  @MaxLength(tagMaxlength, {
+    message: tagMaxLengthErrorMessage,
+  })
```

Comment:
I see what you mean, and I agree. The length validation definitely seems like the most problematic, although the Valid check might not matter in the end, because it could be reasoned that because an exact match is done when finding the tag, if the tag is invalid nothing's going to be updated, since all tags in the DB are considered to be valid.